internal/resources/syntheticmonitoring/resource_check_alert.go

```diff
@@ -0,0 +1,226 @@
+	return resourceCheckAlertRead(ctx, d, c)
+}
+
+func resourceCheckAlertRead(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	alerts, err := c.GetCheckAlerts(ctx, checkID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Transform alerts into schema format
+	alertsList := make([]map[string]interface{}, len(alerts))
+	for i, alert := range alerts {
+		alertMap := map[string]interface{}{
+			"name":      alert.Name,
+			"threshold": alert.Threshold,
+		}
+		if alert.Period != "" {
+			alertMap["period"] = alert.Period
+		}
+		alertsList[i] = alertMap
+	}
+
+	if err := d.Set("alerts", alertsList); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceCheckAlertUpdate(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	alerts, err := makeCheckAlerts(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = c.UpdateCheckAlerts(ctx, checkID, alerts)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceCheckAlertRead(ctx, d, c)
+}
+
+func resourceCheckAlertDelete(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Delete all alerts by setting an empty list
+	_, err = c.UpdateCheckAlerts(ctx, checkID, []model.CheckAlert{})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func makeCheckAlerts(d *schema.ResourceData) ([]model.CheckAlert, error) {
+	alertsSet := d.Get("alerts").(*schema.Set)
+	alertsList := alertsSet.List()
+	alerts := make([]model.CheckAlert, len(alertsList))
+
+	for i, alertMap := range alertsList {
+		alertData := alertMap.(map[string]interface{})
+		name := alertData["name"].(string)
+		period, hasPeriod := alertData["period"].(string)
+
+		if name == AlertNameProbeFailedExecutionsTooHigh && !hasPeriod {
+			return nil, fmt.Errorf("period is required when name is %s", AlertNameProbeFailedExecutionsTooHigh)
+		}
+
+		alert := model.CheckAlert{
+			Name:      name,
+			Threshold: alertData["threshold"].(float64),
+		}
+
+		if hasPeriod {
+			alert.Period = period
+		}
+
+		alerts[i] = alert
+	}
+
+	return alerts, nil
+}
+
+func validateCheckExists(ctx context.Context, c *smapi.Client, checkID int64) error {
+	_, err := c.GetCheck(ctx, checkID)
+	return err
+}
```

Comment:
Related with the other comments, I'm not sure we need this explicit method in the client for alerts.